hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java

```diff
@@ -106,10 +107,10 @@ public StandardSqmSelectTranslator(
 			SqlAstCreationContext creationContext) {
 		super( creationContext, queryOptions, fetchInfluencers, domainParameterXref, domainParameterBindings );
 
-		if ( fetchInfluencers != null
-				&& fetchInfluencers.getEffectiveEntityGraph() != null
-				&& fetchInfluencers.getEffectiveEntityGraph().getSemantic() != null ) {
-			this.entityGraphTraversalState = new StandardEntityGraphTraversalStateImpl( fetchInfluencers.getEffectiveEntityGraph() );
+		final AppliedGraph appliedGraph = queryOptions.getAppliedGraph();
+		if (  appliedGraph.getSemantic() != null && appliedGraph.getGraph() != null ) {
+			this.entityGraphTraversalState = new StandardEntityGraphTraversalStateImpl(
+					appliedGraph.getSemantic(), appliedGraph.getGraph() );
```

Comment:
This is the biggest coding bug flushed out. For HQL/Criteria, we should grab entity graph info from `QueryOptions`, not `LoadQueryInfluencers`.